src/plugins/stake-plugins/uniswapV2/policies/VelodromeV2StakePolicy.ts

```diff
@@ -0,0 +1,728 @@
+        const lpTokenBalance = (await eco.multipass(p => lpTokenContract.connect(p).balanceOf(signerAddress))).toString()
+        const assetAmountsFromLp = await lpTokenToAssetPairAmounts(policyInfo, lpTokenBalance)
+        await Promise.all(
+          allocations
+            .filter(allocation => allocation.allocationType === 'stake')
+            .map(async allocation => {
+              const balanceResponse = await (async () => {
+                const isNativeToken = allocation.currencyCode === policyInfo.parentCurrencyCode
+                if (isNativeToken) {
+                  return await eco.multipass(async p => await p.getBalance(signerAddress))
+                }
+                const tokenAContract = eco.makeContract(allocation.currencyCode)
+                return await eco.multipass(p => tokenAContract.connect(p).balanceOf(signerAddress))
+              })()
+              const balanceAmount = fromHex(balanceResponse._hex)
+              const fromLpToken = assetAmountsFromLp[serializeAssetId(allocation)].nativeAmount
+              const totalBalance = add(balanceAmount, fromLpToken)
+              const isBalanceEnough = lte(allocation.nativeAmount, totalBalance)
+              if (!isBalanceEnough) {
+                throw new Error(sprintf(lstrings.stake_error_insufficient_s, allocation.currencyCode))
+              }
+            })
+        )
+
+        // 2. Approve Swap Router contract for every stake token contract (excluding native token)
+        await Promise.all(
+          allocations.map(async allocation => {
+            // We don't need to approve the stake pool contract for the token earned token contract
+            if (allocation.allocationType === 'claim') return
+            // We don't need to approve the native token asset
+            const isNativeToken = allocation.currencyCode === policyInfo.parentCurrencyCode
+            if (isNativeToken) return
+
+            const tokenAContract = eco.makeContract(allocation.currencyCode)
+            const spenderAddress = swapRouterContract.address
+            txs.build(
+              (gasLimit =>
+                async function approveSwapRouter({ signer }) {
+                  const allowanceResult = await tokenAContract.allowance(signer.address, spenderAddress)
+                  if (allowanceResult.gte(allocation.nativeAmount)) return
+
+                  const result = await tokenAContract.connect(signer).approve(spenderAddress, BigNumber.from(allocation.nativeAmount), {
+                    gasLimit,
+                    gasPrice,
+                    nonce: nextNonce()
+                  })
+                  cacheTxMetadata(result.hash, parentCurrencyCode, {
+                    name: metadataName,
+                    category: 'Expense:Fees',
+                    notes: `Approve ${metadataLpName} liquidity pool contract`
+                  })
+                })(gasLimitAcc('100000'))
```

Comment:
Why is this so much more expensive than in `cemetaryPolicy`? Seems like they should be identical since they are approving the token contracts